sdmetrics/reports/multi_table/quality_report.py

```diff
@@ -94,25 +100,27 @@ def generate(self, real_data, synthetic_data, metadata):
                     if 'score' in self._metric_results[metric.__name__]:
                         prop_scores.append(self._metric_results[metric.__name__]['score'])
                     else:
-                        score = np.nanmean(
-                            [
-                                table_breakdown['score'] for _, table_breakdown
-                                in self._metric_results[metric.__name__].items()
-                            ]
-                        )
-                        prop_scores.append(score)
-            else:
-                for metric in metrics:
-                    score = np.nanmean(
-                        [
-                            breakdown['score'] for _, table_breakdowns
+                        metric_scores = [
+                            table_breakdown['score'] for _, table_breakdown
                             in self._metric_results[metric.__name__].items()
-                            for _, breakdown in table_breakdowns.items()
+                            if not np.isnan(table_breakdown['score'])
                         ]
-                    )
-                    prop_scores.append(score)
-
-            self._property_breakdown[prop] = np.nanmean(prop_scores)
+                        if len(metric_scores) > 0:
+                            prop_scores.append(np.mean(metric_scores))
+            else:
+                for metric in metrics:
+                    metric_scores = [
+                        breakdown['score'] for _, table_breakdowns
+                        in self._metric_results[metric.__name__].items()
+                        for _, breakdown in table_breakdowns.items()
+                        if not np.isnan(breakdown['score'])
+                    ]
+                    if len(metric_scores) > 0:
+                        prop_scores.append(np.mean(metric_scores))
+
+            self._property_breakdown[prop] = np.nanmean(prop_scores) if (
```

Comment:
This multiline if/else looks weird. Assuming you can't do it in 2 lines because of lint, maybe just use a normal if/else block.

Reply:
I can fit it in 2 lines. I'll update it to that.